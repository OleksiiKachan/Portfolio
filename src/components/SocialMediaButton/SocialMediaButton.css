.smButton {
    padding: 7px;
    font-size: 8px;
    width: 8px;
    text-align: center;
    text-decoration: none;
    border-radius: 50%;
}

.smButton:hover {
    opacity: 0.7;
}

.smButton_linkedin {
	background: #007bb5;
	color: white;
}

.smButton_facebook {
	background: #3B5998;
	color: white;
}

.smButton_vk {
	background: #45668e;
	color: white;
}

.smButton_instagram {
	background: #125688;
	color: white;
}