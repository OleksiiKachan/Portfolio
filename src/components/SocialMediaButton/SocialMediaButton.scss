.smButton {
  padding: 7px;
  font-size: 8px;
  width: 8px;
  text-align: center;
  text-decoration: none;
  border-radius: 50%;
}

.smButton:hover {
  opacity: 0.7;
}

.smButton__text_hidden {
  display: none;
}

.smButton_linkedin {
  background-color: #007bb5;
  color: white;
}

.smButton_facebook {
  background-color: #3b5998;
  color: white;
}

.smButton_vk {
  background-color: #45668e;
  color: white;
}

.smButton_instagram {
  background-color: #125688;
  color: white;
}

.smButton_github {
  font-size: 23px;
  width: 23px;
  padding: 0;
  padding-top: 3px;
}
