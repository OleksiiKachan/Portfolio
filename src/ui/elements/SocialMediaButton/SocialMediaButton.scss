.socialMediaButton {
  &__icon {
    width: 36px;
    height: 36px;
    display: block;
    background-position: center;
    background-size: contain;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.25);
    }

    &-linkedin {
      background-image: url('../../assets/icons/icon_linkedin-blank.svg');

      &:hover {
        background-image: url('../../assets/icons/icon_linkedin-filled.svg');
      }
    }

    &-github {
      background-image: url('../../assets/icons/icon_github-blank.svg');

      &:hover {
        background-image: url('../../assets/icons/icon_github-filled.svg');
      }
    }

    &-instagram {
      background-image: url('../../assets/icons/icon_instagram-blank.svg');

      &:hover {
        background-image: url('../../assets/icons/icon_instagram-filled.svg');
      }
    }

    &-telegram {
      background-image: url('../../assets/icons/icon_telegram-blank.png');

      &:hover {
        background-image: url('../../assets/icons/icon_telegram-filled.png');
      }
    }

    &-facebook {
      background-image: url('../../assets/icons/icon_facebook-blank.svg');

      &:hover {
        background-image: url('../../assets/icons/icon_facebook-filled.svg');
      }
    }

    &-vk {
      background-image: url('../../assets/icons/icon_vk-blank.svg');

      &:hover {
        background-image: url('../../assets/icons/icon_vk-filled.svg');
      }
    }
  }
}
