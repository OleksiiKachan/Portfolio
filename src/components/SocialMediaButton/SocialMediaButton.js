import React, { Component } from 'react';
import classNames from 'classnames';
import './SocialMediaButton.scss';

/*
    <SocialMediaButton link='' resource='' />
*/
class SocialMediaButton extends Component {
  render() {
    return (
      <a
        href={this.props.link}
        className={classNames(
          'smButton',
          `smButton_${this.props.resource}`,
          'fa',
          `fa-${this.props.resource}`
        )}
        target="_blank"
        rel="noopener noreferrer"
      >
        <span className={classNames('smButton__text_hidden')}>
          {this.props.resource}
        </span>
      </a>
    );
  }
}

export default SocialMediaButton;
