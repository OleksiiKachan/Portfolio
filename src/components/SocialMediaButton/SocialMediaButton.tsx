import React, { Component } from 'react';
import classNames from 'classnames';
import './SocialMediaButton.scss';

/*
    <SocialMediaButton link='' resource='' />
*/
class SocialMediaButton extends Component<any, any> {
  render() {
    return (
      <a
        href={this.props.link}
        className={classNames(
          'smButton',
          `smButton_${this.props.resource}`,
          'fa',
          `fa-${this.props.resource}`
        )}
        target="_blank"
        rel="noopener noreferrer"
      >
        <span className={classNames('smButton__text_hidden')}>
          {this.props.resource}
        </span>
      </a>
    );
  }
}

export default SocialMediaButton;
