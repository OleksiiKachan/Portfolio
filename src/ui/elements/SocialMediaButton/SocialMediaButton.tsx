import React, { CSSProperties } from 'react';
import classNames from 'classnames';
import './SocialMediaButton.scss';

/*
    <SocialMediaButton
      type=''
      url=''
      className=''
      style={}
    />
*/

type PropsType = {
  type: string;
  url: string;
  className?: string;
  style?: CSSProperties;
};

export default ({ type, url, className, style }: PropsType) => {
  return (
    <a
      href={url}
      target="_blank"
      rel="noopener noreferrer"
      className={classNames('socialMediaButton', className)}
      style={style}
    >
      <i
        className={classNames(
          'socialMediaButton__icon',
          `socialMediaButton__icon-${type}`
        )}
      />
    </a>
  );
};
