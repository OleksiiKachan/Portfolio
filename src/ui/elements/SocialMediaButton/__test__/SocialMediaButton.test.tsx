import React from 'react';
import { shallow } from 'enzyme';
import SocialMediaButton from '../SocialMediaButton';

describe('SocialMediaButton', () => {
  it('Snapshot', () => {
    const component = shallow(
      <SocialMediaButton type="github" url="github.com" />
    );

    expect(component).toMatchSnapshot();
  });
});
