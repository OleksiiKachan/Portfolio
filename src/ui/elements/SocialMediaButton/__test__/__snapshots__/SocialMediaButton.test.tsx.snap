// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaButton Snapshot 1`] = `
<a
  className="socialMediaButton"
  href="github.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <i
    className="socialMediaButton__icon socialMediaButton__icon-github"
  />
</a>
`;
